### PR TITLE
Check for incomparable precedences

### PR DESCRIPTION
**Pull request by lukaszcz**
Give an error when there are two operators with incomparable precedences instead of linearising the ordering arbitrarily.

### CODE REVIEW

File: src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs [hunk 1]
```diff
@@ -2282,6 +2294,60 @@ resolveSyntaxDef = \case
...
+
+checkExpressionPrecedences :: Members '[Error ScoperError, InfoTableBuilder] r => ExpressionAtoms 'Scoped -> Sem r ()
+checkExpressionPrecedences (ExpressionAtoms atoms _) =
+  checkPrecedences opers
+  where
+    opers :: [S.Name]
+    opers = mapMaybe getIden (toList atoms)
+
+    getIden :: ExpressionAtom 'Scoped -> Maybe S.Name
+    getIden a = case a of
```
Comment (jonaprieto) [inline src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs:2333]:
```suggestion
    getIden = \case
```

File: src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs [hunk 2]
```diff
@@ -2282,6 +2294,60 @@ resolveSyntaxDef = \case
...
+  where
+    opers :: [S.Name]
+    opers = mapMaybe getIden (toList atoms)
+
+    getIden :: PatternAtom 'Scoped -> Maybe S.Name
+    getIden = \case
+      PatternAtomIden i -> case i of
+        PatternScopedConstructor c -> Just c
+        _ -> Nothing
+      _ -> Nothing
```
Comment (jonaprieto) [inline src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs:2349]: Move this to `src/Juvix/Compiler/Concrete/Extra.hs`

File: src/Juvix/Data/Fixity.hs [hunk 3]
```diff
@@ -68,13 +70,13 @@ isUnary :: Fixity -> Bool
...
-appFixity = Fixity PrecApp (Binary AssocLeft)
+appFixity = Fixity PrecApp (Binary AssocLeft) Nothing
 
 funFixity :: Fixity
-funFixity = Fixity PrecArrow (Binary AssocRight)
+funFixity = Fixity PrecArrow (Binary AssocRight) Nothing
 
 updateFixity :: Fixity
-updateFixity = Fixity PrecUpdate (Unary AssocPostfix)
+updateFixity = Fixity PrecUpdate (Unary AssocPostfix) Nothing
```
Comment (jonaprieto) [inline src/Juvix/Data/Fixity.hs:79]: Change this to follow the pattern `= Fixity {_fixityPrecedence = .., _fixityArity=..`.

File: tests/Compilation/positive/test060.juvix [hunk 4]
```diff
@@ -17,7 +17,8 @@ type Pair (A B : Type) :=
     };
 
 mf : Pair (Pair Bool (List Nat)) (List Nat) → Bool
-  | mkPair@{fst := mkPair@{fst; snd := nil}; snd := zero :: _} := fst
```
Comment (jonaprieto) [inline tests/Compilation/positive/test060.juvix:20]: These changes are not related to the topic of this PR. Undo this.

**Review by jonaprieto — CHANGES_REQUESTED** [review]
Please resolve the comments/suggestions.
The rest looks good to me.

File: tests/Compilation/positive/test060.juvix (hunk 4 above)
Comment (lukaszcz) [reply tests/Compilation/positive/test060.juvix:20]: Yes, I will. It's the work of the formatter.

File: src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs (hunk 1 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs]: done

File: src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs (hunk 2 above)
Comment (lukaszcz) [reply src/Juvix/Compiler/Concrete/Translation/FromParsed/Analysis/Scoping.hs]: done

File: src/Juvix/Data/Fixity.hs (hunk 3 above)
Comment (lukaszcz) [reply src/Juvix/Data/Fixity.hs]: done

File: tests/Compilation/positive/test060.juvix (hunk 4 above)
Comment (lukaszcz) [reply tests/Compilation/positive/test060.juvix]: done
